sentry/src/routes/analytics.rs

```diff
@@ -0,0 +1,97 @@
+use crate::db::analytics::get_analytics;
+use crate::success_response;
+use crate::Application;
+use crate::ResponseError;
+use crate::RouteParams;
+use crate::Session;
+use hyper::{Body, Request, Response};
+use primitives::adapter::Adapter;
+use primitives::analytics::AnalyticsQuery;
+use redis::aio::MultiplexedConnection;
+
+pub async fn publisher_analytics<A: Adapter>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    process_analytics(req, app, false, false)
+        .await
+        .map(success_response)
+}
+
+pub async fn analytics<A: Adapter>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let request_uri = req.uri().to_string();
+    let redis = app.redis.clone();
+
+    match redis::cmd("EXISTS")
+        .arg(&request_uri)
+        .query_async::<_, String>(&mut redis.clone())
+        .await
+    {
+        Ok(response) => Ok(success_response(response)),
```

Comment:
`Exists` returns a string representation of either `0` or `1` see https://redis.io/commands/exists#return-value

Which means that you probably should probably check for that otherwise the next matching arm doesn't make any sense as it will only work if there is an Error that occurred trying to reach `redis` or executing the command.

Edit: Apparently this should be a `GET`, so keep in mind the return values of `GET`, see https://redis.io/commands/get#return-value